src/ts/index.js

```diff
@@ -0,0 +1,153 @@
+'use strict'
+
+const path = require('path')
+const execa = require('execa')
+const fs = require('fs-extra')
+const rimraf = require('rimraf')
+const merge = require('merge-options')
+const { fromRoot, fromAegir, repoDirectory, hasFile } = require('../utils')
+let baseTsConfig = require('./../config/aegir-tsconfig.json')
+
+if (hasFile('tsconfig.json')) {
+  baseTsConfig = require(fromRoot('tsconfig.json'))
+}
+
+module.exports = (argv) => {
+  const forwardOptions = argv['--'] ? argv['--'] : []
+
+  if (argv.preset === 'check') {
+    return check(forwardOptions)
+  }
+
+  if (argv.preset === 'types') {
+    return types(forwardOptions)
+  }
+
+  if (argv.preset === 'types-clean') {
+    return typesClean()
+  }
+
+  if (argv.preset === 'docs') {
+    return docs(forwardOptions)
+  }
+
+  if (!argv.preset) {
+    return execa('tsc', [
+      ...forwardOptions
+    ], {
+      localDir: path.join(__dirname, '../..'),
+      preferLocal: true,
+      stdio: 'inherit'
+    })
+  }
+}
+
+const check = async (forwardOptions) => {
+  const configPath = fromRoot('tsconfig-check.json')
```

Comment:
I think it would be better to fail if there is no `tsconfig.json` (which can include single [`extends`](https://www.typescriptlang.org/tsconfig#extends) directive e.g. `{ "extends": "aegir/src/config/aegir-tsconfig.json"}`) than do this. Otherwise we introduce another instance of other tools like vscode making different assumptions than tools than aegir.